packages/core/chrome/core-chrome-browser-internal/src/ui/project/app_menu.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiThemeProvider, useEuiTheme } from '@elastic/eui';
+import { css } from '@emotion/react';
+import { MountPoint } from '@kbn/core-mount-utils-browser';
+import React from 'react';
+import { HeaderActionMenu } from '../header/header_action_menu';
+import { SIZE_COLLAPSED, SIZE_EXPANDED } from './navigation';
+
+interface AppMenuBarProps {
+  isOpen: boolean;
+  headerActionMenuMounter: { mount: MountPoint<HTMLElement> | undefined };
+}
+export const AppMenuBar = ({ isOpen, headerActionMenuMounter }: AppMenuBarProps) => {
+  const { euiTheme } = useEuiTheme();
+  return (
+    <EuiThemeProvider>
+      <div
+        className="header__actionMenu"
+        data-test-subj="kibanaProjectHeaderActionMenu"
+        css={css`
+          /* span the width of the body content (viewport - width of side nav) */
+          width: calc(100% - ${isOpen ? SIZE_EXPANDED : SIZE_COLLAPSED}px);
+        `}
+      >
+        <EuiFlexGroup
+          css={css`
+            /* EuiPanel-like styling */
+            padding: ${euiTheme.size.s};
```

Comment:
Just curious - is the header bar supposed to be a specific width, e.g. 48px high? If so, only setting the padding means that its height is dependent on the content of the bar, which means it can possibly be either too tall or too short if devs are passing in totally wildcard content.

If the content of `<HeaderActionMenu />` is always a set of buttons on the correct height, then no worries, you can disregard this concern :)

Reply:
I believe the app menu bar is supposed to be fixed, using the `$kbnHeaderBannerHeight` variable. I think we can disregard the concern. I follow what you are saying though: today, the Observability solution mounts an empty div into this bar, which creates an app menu bar that is empty and only has the padding:

<img width="1634" alt="pfng" src="https://github.com/elastic/kibana/assets/908371/0058e333-fd3d-4d44-bb09-65ccf5667f22">

Reply:
Ideally, I imagine it would be most scalable solution would be to just use padding (rather than using a fixed height) and communicate to consumers that the majority of situations should use the small button sizes. However, if/when there are situations that require different sized buttons or some kind of alternative content beyond buttons in the toolbar, the avoidance of using a fixed height would allow to toolbar to scale as needed. CCing @ryankeairns in case he has any thoughts on the matter as well.

Reply:
If the toolbar height is meant to be flexible, awesome! 👍 Just wasn't sure that was the case, as that's a departure from the previous headers.

Reply:
Hi, sorry I think I got a bit lost in this conversation. The app menu bar has a fixed height. This is done to ensure that the page content begins at the right spot. There's a hardcoded `margin-bottom` on the `header` element to set the spot where content begins.

If the height was dynamic, I don't really know how to ensure that the page content will flow at the right spot. Since the app menu bar has a fixed position and breaks outside of the flow of other elements, there's sort of a battle against having page content shifted up and appear behind the app menu bar.

Reply:
I'm personally ambivalent as to whether the app menu bar should have a fixed or flexible height. Dev-wise, a fixed height is easier, but design-wise, there may be reasons for a variable height. @MichaelMarcialis would be able to say for sure.